### ISSUE TITLE
Ping System

### ISSUE THREAD

**Issue by TheGlitch76** [FabricMC/fabric-discord-bot#21]
Since K9 has been removed we could use a replacement for its ping system.
As a reminder, the ping system for k9 was essentially that you could register an arbitrary regex, which k9 would run on every message and if it matched send you a DM with a link to the message that triggered it.

I would recommend the following restrictions, compared to what k9 had:
- Only works in channels that the user (maybe even `@everyone` just to be safe?) can read
- Pings are essentially just `input.toLowerCase().contains(ping.toLowerCase())`, no regex involved
- Pings should have a minimum length and/or a maximum amount of triggers per unit of time, to avoid situations where someone wants a ping any time the letters `e` or `t` are used in a message, or is otherwise burdening the bot.

**Comment by gdude2002** [comment]
This is actually a fairly decent idea, honestly. I was going to use RE2 for safer regexes, but that'd still be fairly burdensome — and it's pretty much only a niche requirement.

This will have to come after the database rewrite, though.

**Comment by ramidzkh** [comment]
To be honest, I disagree with the restrictions proposed. A regex expression has proven to work with the previous bot, even if people set ridiculous pings such as `/./`. Also, although a cool-off period may seem appealing, it would have to be configurable by users, but that that point its easier to just send every message through. I'd rather have 7 pings overnight and reply to each one, than only get 2 and miss the other 5

**Comment by modmuss50** [comment]
I dont think we really need this feature, feel free to disagree with me, but I feel like it was used to "alert" people to possible drama/whatever so they can chime in their view.

**Comment by ramidzkh** [comment]
Most people has pings for their names, or projects they worked on.

![image](https://user-images.githubusercontent.com/19631324/99443356-d1ec5580-296e-11eb-86d9-ba33e5898b00.png)
In this extract of the original pings dump, there's only two which looks kind of suspicious, but nothing drama related

**Comment by TheGlitch76** [comment]
My intention wasn't for only getting two pings if your project was
mentioned a bunch, but instead as a filter to avoid abuse. The idea is that
to 95% of the users it feels exactly the same

**Comment by SilverAndro** [comment]
The main possible issue I see with it is that inconsistent spacing may mean you miss out on someone mentioning something

If I have project A B-C but someone says AB-C, I don't receive a ping, and trying to build the "simple" system to accommodate all the possible variations is bound to lead to issues

With a regex based system I can simply match 0 or more spaces and not miss any messages that should ping, or similar misspellings

**Comment by dexman545** [comment]
I agree with ramid, the system has the most use in knowing when projects you work on are being discussed - like if a user has a question about something. Fabric isn't so small everyone knows who did what project and can send a ping their way when they are needed.

If the fear is drama, you could make it more curated, but I see no reason to increase the moderation burden and place a barrier to the system.
